witness/tests/Test/Dscp/Witness/Explorer/Explorer.hs

```diff
@@ -119,20 +147,26 @@ editSomeKV mp action = do
 spoilFairCV
     :: FairCV
     -> Gen FairCV
-spoilFairCV (FairCV cv) = do
+spoilFairCV (FairCV sAddr sName cv) = do
     var <- arbitraryBoundedEnum
-    fmap FairCV $ editSomeKV cv $ case var of
-        WrongAddr -> \oldAddr subCv ->
-            (,subCv) <$> arbitrary `suchThat` (/= oldAddr)
-        WrongHeaderHash -> \addr subCv -> do
-            subCv' <- editSomeKV subCv $ \oldHash proof ->
-                (,proof) <$> arbitrary `suchThat` (/= oldHash)
-            return (addr, subCv')
-        WrongProof -> \addr subCv -> do
-            subCv' <- editSomeKV subCv $ \hhash proof -> do
-                proof' <- spoilMerkleProof proof
-                return (hhash, proof')
-            return (addr, subCv')
+    case var of
+        WrongStudent ->
+            (\addr -> FairCV addr sName cv) <$>
+            arbitrary `suchThat` (/= sAddr)
+        _ -> fmap (FairCV sAddr sName) $
+             editSomeKV cv $ case var of
+            WrongAddr -> \oldAddr subCv ->
+                (,subCv) <$> arbitrary `suchThat` (/= oldAddr)
+            WrongHeaderHash -> \addr subCv -> do
+                subCv' <- editSomeKV subCv $ \oldHash proof ->
+                    (,proof) <$> arbitrary `suchThat` (/= oldHash)
+                return (addr, subCv')
+            WrongProof -> \addr subCv -> do
+                subCv' <- editSomeKV subCv $ \hhash proof -> do
+                    proof' <- spoilMerkleProof proof
+                    return (hhash, proof')
+                return (addr, subCv')
+            _ -> error "already ruled out"
```

Comment:
https://t3.ftcdn.net/jpg/01/66/72/38/240_F_166723883_K6GfLEV5FJj6iWGhIENbG1llkzpwjRlt.jpg

Reply:
Okay, that indeed has become mildly obscure. I'll make this prettier.

Reply:
:thx: